pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -173,9 +178,50 @@ func (r *InferenceGraphReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return reconcile.Result{}, errors.Wrapf(err, "fails to create DeployConfig")
 	}
 
+	// examine DeletionTimestamp to determine if object is under deletion
+	if graph.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer.
+		if !utils.Includes(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName) {
+			graph.ObjectMeta.Finalizers = append(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(graph.ObjectMeta.Finalizers, ",") + "]"
+			patchJson, _ := yaml.YAMLToJSON([]byte(patchYaml))
+			if err = r.Patch(ctx, graph, client.RawPatch(types.MergePatchType, patchJson)); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if utils.Includes(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName) {
+			// our finalizer is present, so lets handle any external dependency
+			if err = r.onDeleteCleanup(ctx, graph); err != nil {
+				// if fail to delete the external dependency here, return with error
+				// so that it can be retried
+				return ctrl.Result{}, err
+			}
+
+			// remove our finalizer from the list and update it.
+			graph.ObjectMeta.Finalizers = utils.RemoveString(graph.ObjectMeta.Finalizers, constants.InferenceGraphFinalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(graph.ObjectMeta.Finalizers, ",") + "]"
+			patchJson, _ := yaml.YAMLToJSON([]byte(patchYaml))
+			if err = r.Patch(ctx, graph, client.RawPatch(types.MergePatchType, patchJson)); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
 	deploymentMode := isvcutils.GetDeploymentMode(graph.Status.DeploymentMode, graph.ObjectMeta.Annotations, deployConfig)
 	r.Log.Info("Inference graph deployment ", "deployment mode ", deploymentMode)
 	if deploymentMode == constants.RawDeployment {
+		// If the inference graph has auth enabled, create the supporting resources
+		err = handleInferenceGraphRawAuthResources(ctx, r.Clientset, r.Scheme, graph)
```

Comment:
So this is for raw only. Is there some logic that will be added for serverless?

Reply:
I am confused about the GA for IG. IG will be supporting both modes at the same time or one by one?
Raw would be the first target?

Reply:
> both modes at the same time or one by one?

Both modes at the same time.

> So this is for raw only. Is there some logic that will be added for serverless?

Yes. For Serverless auth is handled like in ISVCs via authorino. Such logic was added in odh-model-controller some weeks ago: https://github.com/opendatahub-io/odh-model-controller/pull/345

Reply:
ok thx